File: tarok_java/Igralec.java
import java.util.Vector;

public class Igralec {
    public static String Ime = "JavaIgralec";
    Vector<Karta> karte;

    public Igralec() {
    }

    public TipIgre zacniIgro(Vector<Karta> karteZaVRoko) {
		return new TipIgre(3, false, Barva.KARO);
    
    }
    
    public void zalozi(Vector<Karta> karte, Vector<Vector<Karta>> talon, Vector<Karta> vzete, Vector<Karta> zalozene) {
    	vzete.addAll(talon.get(1));
    	for (int i=0;i<karte.size();i++) {
    		if (karte.get(i).vrednost != 14 || karte.get(i).barva == Barva.TAROK) zalozene.add(karte.get(i));
    		if (zalozene.size() == 3) break;
    	}
    }

    public void zacniRedniDel(int idxIgralca, int glavniIgralec, TipIgre tipIgre, Vector<Karta> ostanekTalona, Vector<Karta> pobraneKarteTalona) {
    	
    }
    
    public Karta vrziKarto(Vector<Karta> mojeKarte, Vector<Karta> karteNaMizi, int prviIgralec) {
		Vector<Karta> veljavne = Uporabno.veljavnePoteze(mojeKarte, karteNaMizi);
		return veljavne.get(0);
    }
    
    public void konecKroga(boolean zmagal, int prviIgralecVKrogu, int zmagovalec, Vector<Karta> karteNaMizi) {
    }
    
    public void KonecIgre(String razlog) {
        System.out.println("Konec igre, razlog = " + razlog);
    }
}

File: tarok_py/igralec.py
from common import *

class Igralec:
	ime = "Dumb"

	def __init__(self):
		pass

	def zacniIgro(self, karteZaVRoko):
		# zapomnimo si svoje karte v kaksno spremenljivko
		self.karte = karteZaVRoko
		# vedno poskusimo igrati trojko v kari
		return TipIgre(3, False, KARO)
		
	def zalozi(self, karte, talon):
		# najdemo tri karte (ker smo sli igrat trojko), ki jih lahko zalozimo (= ki niso kralj):
		zalozeneKarte = [k for k in karte if (k.vrednost!=14 or k.barva==TAROK)][:3]
		# vedno izberemo drugi kupcek iz talona
		return (talon[1], zalozeneKarte)
		
	def zacniRedniDel(self, idxIgralca, glavniIgralec, tipIgre, ostanekTalona, pobraneKarteTalona):
		# resen igralec dobi tukaj kup informacij, ki si jih je vredno zapomniti
		return
		
	def vrziKarto(self, mojeKarte, karteNaMizi, prviIgralec):
		# vrzemo prvo dopustno karto
		return veljavnePoteze(mojeKarte, karteNaMizi)[0]
		
	def konecKroga(self, zmagal, prviIgralecVKrogu, zmagovalec, karteNaMizi):
		# resen igralec si morda zeli zapomniti, katere karte so padle.
		return
		
	def konecIgre(self, razlog):
		# izpisemo razlog, da vidimo, ce je slo vse po nacrtih
		print 'KONEC IGRE.', razlog
